base/char.jl

```diff
@@ -219,8 +222,40 @@ hash(x::AbstractChar, h::UInt) = hash(Char(x), h)
 widen(::Type{T}) where {T<:AbstractChar} = T
 
 @inline -(x::AbstractChar, y::AbstractChar) = Int(x) - Int(y)
-@inline -(x::T, y::Integer) where {T<:AbstractChar} = T(Int32(x) - Int32(y))
-@inline +(x::T, y::Integer) where {T<:AbstractChar} = T(Int32(x) + Int32(y))
+@inline function -(x::T, y::Integer) where {T<:AbstractChar}
+    if x isa Char
```

Comment:
Would this be better as a separate method?

Reply:
seemed more awkward

Reply:
Measured that 2/3rds of this code was actually always slower, so deleted it, and fixed numerous edge cases in the remaining code. It is about 10% faster to have this code across a benchmark like:
```
@btime '0' + 1
@btime '0' + $(Ref(1))[]
@btime $(Ref('0'))[] + 1
@btime $(Ref('0'))[] + $(Ref(1))[]
```